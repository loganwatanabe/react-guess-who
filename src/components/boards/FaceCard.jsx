import React, {useState, useEffect} from 'react';
import {Paper, Card, CardMedia, CardActionArea, CardContent, Typography, ButtonBase, Grid} from '@material-ui/core'


function FaceCard(props){

	// const [error, setError] = useState(null);
	// const [isLoaded, setIsLoaded] = useState(false);
	const [active, setActive] = useState(true);

	// Note: the empty deps array [] means
	// this useEffect will run once
	// similar to componentDidMount()
	useEffect(() => {
  	})

  	const cardClick = (event)=>{
  		if(props.onClick){
  			props.onClick()
  		}else{
  			setActive(!active)
  		}
  	}

  	const displayImage = ()=>{
  		if(props.data && props.data.url){
  			return(<img src={props.data.url} alt={props.data.name}
          style={{width:"100%", height:"100%", objectFit:"cover"}}/>)
  		}else{
  			return(
  		    <div style={{height: "100%", width: "100%", backgroundColor: "grey", marginLeft: "auto", marginRight: "auto"}} />
  			)
  		}
  	}

    const GridChecker = ({ children }) => {
      if(props.chosen){
        return(children)
      }else{
        return(<Grid item xs={6} sm={4} md={2} style={{minHeight: "360px"}}>{children}</Grid>)
      }
    }

  return (
    <GridChecker>
	  	<Card style={{width: "100%", height: "100%", textAlign: 'center'}} disabled={!active} raised={active}>
	      <CardActionArea style={{height: "100%"}} onClick={cardClick}>
          <div style={{height: "90%", width: "100%", margin: "auto", overflow: "hidden"}}>
            {displayImage()}
          </div>
	        <CardContent style={{height:"10%", padding: "0px 6px"}}>
	        	<Typography gutterBottom variant="h5" component="h2">
	        		{props.data ? props.data.name : "???"}
	        	</Typography>
	        </CardContent>
	        {active ? "" : <div style={{height: '100%', width: '100%', position: 'absolute', top: 0, left: 0, backgroundColor: 'black', opacity: '75%'}}></div>}
	      </CardActionArea>

	    </Card>
    </GridChecker>
  );
}

export default FaceCard;