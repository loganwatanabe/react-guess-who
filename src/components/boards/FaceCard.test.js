// Link.react.test.js
import React from 'react';
import FaceCard from './FaceCard';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <FaceCard name="????"/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});