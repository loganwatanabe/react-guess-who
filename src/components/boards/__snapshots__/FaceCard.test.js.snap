// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "margin": "4px auto 4px",
    }
  }
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation8 MuiPaper-rounded"
    disabled={false}
    style={
      Object {
        "maxHeight": 240,
        "textAlign": "center",
        "width": 240,
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        style={
          Object {
            "fontSize": 24,
            "height": 120,
            "width": 120,
          }
        }
      >
        Draw a Card
      </div>
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          ???
        </h2>
      </div>
      
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</div>
`;
